Source/panels/charpanel.cpp

```diff
@@ -0,0 +1,315 @@
+}
+
+void DrawShadowString(const Surface &out, Point pos, panelEntry &entry)
+{
+	if (entry.label == "")
+		return;
+
+	std::string text_tmp = entry.label;
+	char buffer[32];
+	int spacing = entry.labelSpacing;
+	strcpy(buffer, text_tmp.c_str());
+	if (entry.labelLength > 0)
+		WordWrapGameString(buffer, entry.labelLength, GameFontSmall, spacing);
+	std::string text(buffer);
+	int width = GetLineWidth(text, GameFontSmall, spacing);
+	if (entry.centered)
+		width = entry.length;
+	else
+		pos.x -= width;
+
+	Point finalPos = { pos + entry.position + Displacement { 0, 17 } + entry.labelOffset };
+	UiFlags style = UiFlags::AlignRight;
+	if (entry.centered) {
+		style = UiFlags::AlignCenter;
+		finalPos += Displacement { 7, 0 }; // left border
+	}
+	DrawString(out, text, { finalPos + Displacement { -2, 2 }, { width, 0 } }, style | UiFlags::ColorBlack, spacing, 10);
+	DrawString(out, text, { finalPos, { width, 0 } }, style | UiFlags::ColorSilver, spacing, 10);
+}
+
+void LoadCharPanel()
+{
+	LoadArt("data\\charbg.pcx", &PanelFull);
+	LoadArt("data\\boxleftend26.pcx", &PanelParts[0]);
+	LoadArt("data\\boxmiddle26.pcx", &PanelParts[1]);
+	LoadArt("data\\boxrightend26.pcx", &PanelParts[2]);
+	LoadArt("data\\boxleftend27.pcx", &PanelParts[3]);
+	LoadArt("data\\boxmiddle27.pcx", &PanelParts[4]);
+	LoadArt("data\\boxrightend27.pcx", &PanelParts[5]);
+
+	Point pos = GetPanelPosition(UiPanels::Character, { 0, 0 });
+	const Surface out(PanelFull.surface.get());
+
+	for (auto &entry : panelEntries) {
+		if (entry.statDisplayFunc != nullptr) {
+			if (entry.high)
+				DrawPanelFieldHigh(out, pos + entry.position, entry.length);
+			else
+				DrawPanelFieldLow(out, pos + entry.position, entry.length);
+		}
+		DrawShadowString(out, pos, entry);
+	}
+
+	for (auto &gfx : PanelParts) {
+		gfx.Unload();
+	}
+}
+
+bool CharPanelLoaded = false;
+
+void DrawStatButtons(const Surface &out)
+{
+	if (myPlayer._pStatPts > 0) {
+		if (myPlayer._pBaseStr < myPlayer.GetMaximumAttributeValue(CharacterAttribute::Strength))
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 159 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Strength)] ? 3 : 2);
+		if (myPlayer._pBaseMag < myPlayer.GetMaximumAttributeValue(CharacterAttribute::Magic))
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 187 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Magic)] ? 5 : 4);
+		if (myPlayer._pBaseDex < myPlayer.GetMaximumAttributeValue(CharacterAttribute::Dexterity))
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 216 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Dexterity)] ? 7 : 6);
+		if (myPlayer._pBaseVit < myPlayer.GetMaximumAttributeValue(CharacterAttribute::Vitality))
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 244 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Vitality)] ? 9 : 8);
+	}
+}
+
+void DrawPcxChr(const Surface &out)

```

Comment:
Since it replaces the original one:
```suggestion
void DrawChr(const Surface &out)
```